include/util/opening_hours.hpp

```diff
@@ -0,0 +1,645 @@
+               >> -date_offset
+               >> '-'
+               >> date_to[ph::bind(&OpeningHours::MonthdayRange::to, _val) = _1]
+               >> -date_offset)
+            | (date_from[ph::bind(&OpeningHours::MonthdayRange::from, _val) = _1]
+               >> -(date_offset
+                    >> -lit('+')[ph::bind(&OpeningHours::MonthdayRange::from, _val) = ph::construct<OpeningHours::Monthday>(-1)]
+                    ))
+            ;
+
+        date_offset = (plus_or_minus >> wday) | day_offset;
+
+        date_from
+            = ((-year[_a = _1] >> ((month[_b = _1] >> -daynum[_c = _1]) | daynum[_c = _1]))
+               | variable_date)
+            [_val = ph::construct<OpeningHours::Monthday>(_a, _b, _c)]
+            ;
+
+        date_to
+            = date_from[_val = _1]
+            | daynum[_val = ph::construct<OpeningHours::Monthday>(0, 0, _1)]
+            ;
+
+        variable_date = lit("easter");
+
+        // Year selector
+        year_selector = (year_range % ',')[ph::bind(&OpeningHours::monthdays, _r1) = _1];
+
+        year_range
+            = year[ph::bind(&OpeningHours::MonthdayRange::from, _val) = ph::construct<OpeningHours::Monthday>(_1)]
+            >> -(('-' >> year[ph::bind(&OpeningHours::MonthdayRange::to, _val) = ph::construct<OpeningHours::Monthday>(_1)]
+                  >> -('/' >> uint_))
+                 | lit('+')[ph::bind(&OpeningHours::MonthdayRange::to, _val) = ph::construct<OpeningHours::Monthday>(-1)]);
+
+        // Basic elements
+        plus_or_minus = lit('+')[_val = true] | lit('-')[_val = false];
+
+        hour = uint2_p[_pass = bind([](unsigned x) { return x <= 24; }, _1), _val = _1];
+
+        extended_hour = uint2_p[_pass = bind([](unsigned x) { return x <= 48; }, _1), _val = _1];
+
+        minute = uint2_p[_pass = bind([](unsigned x) { return x < 60; }, _1), _val = _1];
+
+        hour_minutes =
+            hour[_a = _1] >> ':' >> minute[_val = ph::construct<OpeningHours::Time>(_a, _1)];
+
+        extended_hour_minutes = extended_hour[_a = _1] >> ':' >>
+                                minute[_val = ph::construct<OpeningHours::Time>(_a, _1)];
+
+        wday.add
+            ("Su", 0)
+            ("Mo", 1)
+            ("Tu", 2)
+            ("We", 3)
+            ("Th", 4)
+            ("Fr", 5)
+            ("Sa", 6)
+            ;
+
+        daynum
+            = uint2_p[_pass = bind([](unsigned x) { return 01 <= x && x <= 31; }, _1), _val = _1]
+            >> (&~lit(':') | eoi)
+            ;
+
+        weeknum = uint2_p[_pass = bind([](unsigned x) { return 01 <= x && x <= 53; }, _1), _val = _1];
+
+        month.add
+            ("Jan", 1)
+            ("Feb", 2)
+            ("Mar", 3)
+            ("Apr", 4)
+            ("May", 5)
+            ("Jun", 6)
+            ("Jul", 7)
+            ("Aug", 8)
+            ("Sep", 9)
+            ("Oct", 10)
+            ("Nov", 11)
+            ("Dec", 12)
```

Comment:
Re. month and weekday - I hope they're language agnostic, e.g. german mappers also use "Dec" and not "Dez". Reading the spec this should be the case.